openfl/utilities/utils.py

```diff
@@ -170,3 +171,26 @@ def split_tensor_dict_for_holdouts(logger, tensor_dict,
     holdout_tensors = {**holdout_tensors, **not_supported_tensors_dict}
 
     return tensors_to_send, holdout_tensors
+
+
+def sha384sum(file_path):
+    """Calculate SHA384 hash for file specified.
+
+    Args:
+        file_path(path-like): path-like object giving the pathname
+            (absolute or relative to the current working directory)
+            of the file to be opened or an integer file descriptor of the file to be wrapped.
+
+    Returns:
+        str: SHA384 hash value.
+    """
+    h = hashlib.sha384()
+    with open(file_path, 'rb') as file:
+        while True:
+            # Reading is buffered, so we can read smaller chunks.
+            chunk = file.read(h.block_size)
```

Comment:
It's only 128 bytes. It look's like a very small buffer size for reading. It can severely degrade performance. It should be checked, but values are more commonly used a from 4kb to 64kb.